common.gypi

```diff
@@ -218,7 +219,13 @@
     # Forcibly disable -Werror.  We support a wide range of compilers, it's
     # simply not feasible to squelch all warnings, never mind that the
     # libraries in deps/ are not under our control.
-    'cflags!': ['-Werror'],
+    'conditions': [
+      [ 'error_on_warn=="true" and '
+        '(_target_name!="<(node_lib_target_name)" or '
+          '_target_name!="<(node_core_target_name)")', {
+        'cflags!': ['-Werror'],
+      }],
+    ],
```

Comment:
This doesn't appear to be correct? If `error_on_warn` is not `true` we should still turn off `-Werror` as before.

Reply:
@richardlau Sorry about the delay on this :disappointed: I'll take a look now...